src/backend/access/external/pxffilters.c

```diff
@@ -851,12 +852,51 @@ append_attr_from_var(Var* var, List* attrs)
 	return attrs;
 }
 
+/*
+ * append_attr_from_func_args
+ *
+ * extracts all columns from FuncExpr into attrs
+ * assigns false to expressionIsSupported if at least one of items is not supported
+ */
+static List*
+append_attr_from_func_args(FuncExpr *expr, List* attrs, bool* expressionIsSupported) {
+	ListCell *lc = NULL;
+	if (!expressionIsSupported) {
+		return NIL;
+	}
+	foreach (lc, expr->args)
+	{
+		Node *node = (Node *) lfirst(lc);
+		if (IsA(node, FuncExpr)) {
+			attrs = append_attr_from_func_args((FuncExpr *) node, attrs, expressionIsSupported);
+		} else if (IsA(node, Var)) {
+			attrs = append_attr_from_var((Var *) node, attrs);
+		} else if (IsA(node, OpExpr)) {
+			attrs = get_attrs_from_expr((OpExpr *) node, expressionIsSupported);
+		} else {
+			*expressionIsSupported = false;
+			return NIL;
+		}
+	}
+
+	return attrs;
+
+}
+
+/*
+ * get_attrs_from_expr
+ *
+ * extracts and returns list of all columns from Expr
+ * assigns false to expressionIsSupported if at least one of items is not supported
+ */
 static List*
-get_attrs_from_expr(Expr *expr)
+get_attrs_from_expr(Expr *expr, bool* expressionIsSupported)
```

Comment:
an alternative way is to have all these functions return a boolean, indicating whether selection can be made or not  and take a list as an OUT function argument. This feels more intuitive, IMO.

Reply:
I agree

Reply:
Right, might be more intuitive to reverse signatures, but I am not sure if that refactoring will give many benefits other than the more intuitive perception of code, which also makes sense, but I would stick to the current signature at this point.